scripts/generate-changelog.sh

```
@@ -0,0 +1,307 @@
+    local current_version=$1
+    local previous_version=$2
+    
+    print_status "Generating commit history from $previous_version to $current_version..." >&2
+    
+    # Get commit range
+    local commit_range=""
+    if [ "$previous_version" = "0.0.0" ]; then
+        commit_range="HEAD"
+    else
+        # Try to find the commit where previous version was set
+        local previous_commit=$(git log --grep="version.*$previous_version" --oneline -1 --format="%H" 2>/dev/null || echo "")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)